v2/observer/jobs.go

```diff
@@ -203,22 +203,25 @@ func (o *observer) startJobPodTimer(ctx context.Context, pod *corev1.Pod) {
 		pod.Status.Phase != corev1.PodRunning {
 		return
 	}
-	if pod.Annotations[myk8s.AnnotationTimeoutDuration] == "" {
-		return
-	}
 
-	duration := pod.Annotations[myk8s.AnnotationTimeoutDuration]
-	timeout, err := time.ParseDuration(duration)
-	if err != nil {
-		o.errFn(
-			errors.Wrapf(
-				err,
-				"unable to parse timeout duration %q for pod %q",
-				duration,
-				pod.Name,
-			),
-		)
-		return
+	// Default the timeout value based on the Observer's config
+	timeout := o.config.maxJobLifetime
+	// Else, use the value set on the pod itself
+	if pod.Annotations[myk8s.AnnotationTimeoutDuration] != "" {
+		var err error
+		duration := pod.Annotations[myk8s.AnnotationTimeoutDuration]
+		timeout, err = time.ParseDuration(duration)
+		if err != nil {
+			o.errFn(
+				errors.Wrapf(
+					err,
+					"unable to parse timeout duration %q for pod %q",
+					duration,
+					pod.Name,
+				),
+			)
+			return
+		}
 	}
```

Comment:
Shouldn't we be looking at the pod-specified timeout and reducing it if it exceeds the max?

Reply:
Oh, yes!  Thanks.  Will update.

Reply:
Updated in https://github.com/brigadecore/brigade/pull/1442/commits/0561129841bf1e9c850cd6c16c7794c93de1a21a